src/main/java/de/tum/cit/aet/artemis/iris/service/settings/IrisSettingsService.java

```diff
@@ -276,9 +300,135 @@ private IrisCourseSettings updateCourseSettings(IrisCourseSettings existingSetti
         ));
         // @formatter:on
 
+        // Automatically update the exercise settings when the enabledForCategories is changed
+        var newEnabledForCategoriesExerciseChat = existingSettings.getIrisChatSettings() == null ? new TreeSet<String>()
+                : existingSettings.getIrisChatSettings().getEnabledForCategories();
+        if (!Objects.equals(oldEnabledForCategoriesExerciseChat, newEnabledForCategoriesExerciseChat)) {
+            programmingExerciseRepository.findAllWithCategoriesByCourseId(existingSettings.getCourse().getId())
+                    .forEach(exercise -> setEnabledForExerciseByCategories(exercise, oldEnabledForCategoriesExerciseChat, newEnabledForCategoriesExerciseChat));
+        }
+
+        var newEnabledForCategoriesTextExerciseChat = existingSettings.getIrisTextExerciseChatSettings() == null ? new TreeSet<String>()
+                : existingSettings.getIrisTextExerciseChatSettings().getEnabledForCategories();
+        if (!Objects.equals(oldEnabledForCategoriesTextExerciseChat, newEnabledForCategoriesTextExerciseChat)) {
+            textExerciseRepository.findAllWithCategoriesByCourseId(existingSettings.getCourse().getId())
+                    .forEach(exercise -> setEnabledForExerciseByCategories(exercise, oldEnabledForCategoriesTextExerciseChat, newEnabledForCategoriesTextExerciseChat));
+        }
+
         return irisSettingsRepository.save(existingSettings);
     }
 
+    /**
+     * Set the enabled status for an exercise based on it's categories.
+     * Compares the old and new enabled categories, reads the exercise categories,
+     * and updates the Iris chat settings accordingly if the new enabled categories match any of the exercise categories.
+     * This method is used when the enabled categories of the course settings are updated.
+     *
+     * @param exercise                The exercise to update the enabled status for
+     * @param oldEnabledForCategories The old enabled categories
+     * @param newEnabledForCategories The new enabled categories
+     */
+    public void setEnabledForExerciseByCategories(Exercise exercise, SortedSet<String> oldEnabledForCategories, SortedSet<String> newEnabledForCategories) {
+        var removedCategories = new TreeSet<>(oldEnabledForCategories == null ? Set.of() : oldEnabledForCategories);
+        removedCategories.removeAll(newEnabledForCategories);
```

Comment:
Why do you check for null again? You already handled this in the previous method?
If you want to check it here again, you should take a look at code rabbit's comment about a potential null pointer